File: common-lib/src/main/java/club/hosppy/common/error/ServiceException.java
package club.hosppy.common.error;

import club.hosppy.common.api.ResultCode;
import lombok.Getter;

public class ServiceException extends RuntimeException {

    @Getter
    private final ResultCode resultCode;

    public ServiceException(ResultCode resultCode) {
        this.resultCode = resultCode;
    }

    public ServiceException(ResultCode resultCode, Throwable cause) {
        super(cause);
        this.resultCode = resultCode;
    }
}

File: src/main/kotlin/club/hosppy/common/error/ServiceException.kt
package club.hosppy.common.error

import club.hosppy.common.api.ResultCode

class ServiceException : RuntimeException {
    val resultCode: ResultCode

    constructor(resultCode: ResultCode) {
        this.resultCode = resultCode
    }

    constructor(resultCode: ResultCode, cause: Throwable) : super(cause) {
        this.resultCode = resultCode
    }
}